src/modules/dashboard/components/HookedTokenIcon/HookedTokenIcon.tsx

```diff
@@ -44,11 +45,21 @@ const loadTokenImages = async (address: Address): Promise<Response> => {
   return fetch(tokenImageUrl);
 };
 
+const getBase64image = (blob): Promise<any> => {
```

Comment:
Why is this an `async` function, the only async call is done outside of this, when fetching the blob.

Also, this can be simplified as you don't need to set up the `onload` listener.

```js
const fileReader = new FileReader();
fileReader.readAsDataURL(blob);
return fileReader.result;
```

No async at all.

Reply:
With this, you will get only empty string

Reply:
hmmm... I remember testing locally and working

But I'm not saying you're not right :) Maybe I missed something.

I'll check again.